### ISSUE TITLE
False positive on POSIX sh with export -p

### ISSUE THREAD

**Issue by contivero** [koalaman/shellcheck#1432]
#### For bugs
- Rule Id: SC2039
- Shellcheck version 0.6.0 and online. 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2039)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
export -p
```

#### Here's what shellcheck currently says:
```
Line 2:
export -p
       ^-- SC2039: In POSIX sh, export -p is undefined.
```

#### Here's what I wanted or expected to see:
Nothing, `-p` is the only valid flag for export in POSIX sh ([see spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#export)). This false positive was introduced when #1130 was fixed.
